### PR TITLE
Put correct function call for Return Attributes

### PR DESCRIPTION
**Pull request by iserrato**
/cc: @jkoritzinsky @agocke 

Found while writing Code Fix provider for DynamicallyAccessedMembers trimmer warnings

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]
LGTM

**Comment by agocke** [comment]
Needs a test (likely in https://github.com/dotnet/roslyn/blob/61e0850bbeebec2b10e4829bac6b48ecabc02dcf/src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs)

File: src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs [hunk 1]
```diff
@@ -480,5 +481,97 @@ public class C
...
+}
+""";
+
+            var cu = SyntaxFactory.ParseCompilationUnit(code);
+            var cls = cu.Members[0];
+
+            var editor = GetEditor(cu);
+            var methodX = editor.Generator.GetMembers(cls)[0];
+
+            if (methodX is not MethodDeclarationSyntax method)
```
Comment (agocke) [inline src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs:517]: Looks like this shouldn't be necessary.

**Review by agocke — APPROVED** [review]
LGTM aside from minor comment.

**Comment by agocke** [comment]
@CyrusNajmabadi Mind reviewing this small change?

File: src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs (hunk 1 above)
Comment (iserrato) [reply src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs:517]: I need to cast to MethodDeclarationSyntax. Are you saying the check is unneccessary and I should just cast it in its definition?

Comment (agocke) [reply src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs:517]: Yup, I think so. It can't fail, right?

**Comment by agocke** [comment]
Also ping @dotnet/roslyn-ide

File: src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs (hunk 1 above)
Comment (iserrato) [reply src/Workspaces/CoreTest/Editing/SyntaxEditorTests.cs:517]: That's correct. I'll change it to a cast.

**Review by jasonmalinowski — APPROVED** [review]

**Comment by iserrato** [comment]
> ⚠️ There are test failures in this build for pipelines that are also failing in main

This was reported by the build analyzer, the failure reported is also in main.

**Pull request merged**